homeassistant/components/homekit_controller/light.py

```diff
@@ -153,23 +159,36 @@ async def async_turn_on(self, **kwargs: Any) -> None:
         temperature = kwargs.get(ATTR_COLOR_TEMP)
         brightness = kwargs.get(ATTR_BRIGHTNESS)
 
-        characteristics = {}
+        characteristics: dict[str, Any] = {}
 
         if hs_color is not None:
-            characteristics.update(
-                {
-                    CharacteristicsTypes.HUE: hs_color[0],
-                    CharacteristicsTypes.SATURATION: hs_color[1],
-                }
-            )
+            characteristics[CharacteristicsTypes.HUE] = hs_color[0]
+            characteristics[CharacteristicsTypes.SATURATION] = hs_color[1]
 
         if brightness is not None:
             characteristics[CharacteristicsTypes.BRIGHTNESS] = int(
                 brightness * 100 / 255
             )
 
         if temperature is not None:
-            characteristics[CharacteristicsTypes.COLOR_TEMPERATURE] = int(temperature)
+            if self.service.has(CharacteristicsTypes.COLOR_TEMPERATURE):
+                characteristics[CharacteristicsTypes.COLOR_TEMPERATURE] = int(
+                    temperature
+                )
+            else:
+                # Some HomeKit devices implement color temperature with HS
+                # since the spec does not permit the COLOR_TEMPERATURE characteristic
+                # and the HUE and SATURATION characteristics to be present at the
+                # same time.
+                #
+                # In practice there are a lot of devices that have
+                # both. If we have both, we use the COLOR_TEMPERATURE characteristic
+                # and ignore the HS characteristics when setting color temperature.
+                hue_sat = color_util.color_temperature_to_hs(
+                    color_util.color_temperature_mired_to_kelvin(temperature)
+                )
+                characteristics[CharacteristicsTypes.HUE] = hue_sat[0]
```

Comment:
Is it valid for the user to call us with a temp and h/s? What should happen in that case? Because if they do that on a bulb with no temp then this will throw away their h/s in favour of ours? I suspect in practice won't matter much, but wanted to check.

Reply:
Its not valid to call with both, only one or the other

Reply:
8983f397a0b1006f874315bce6f36d525a0a7dee

Reply:
I think its actually possible for them to send both but one has to win

Reply:
9c7dc85053c8ea6d90fdd500e3388b8ae445dfb6

Reply:
I tweaked it a bit more. I'm not sure this would ever happen on purpose though (of course someone will do it though)